pkg/meta/slice.go

```diff
@@ -183,21 +183,14 @@ func compactChunk(ss []*slice) (uint32, uint32, []Slice) {
 func skipSome(chunk []*slice) int {
 	var skipped int
 	var total = len(chunk)
+	_, size, _ := compactChunk(chunk)
 	for skipped < total {
-		ss := chunk[skipped:]
-		pos, size, c := compactChunk(ss)
-		first := ss[0]
-		if first.len < (1<<20) || first.len*5 < size || size == 0 {
-			// it's too small
-			break
-		}
-		isFirst := func(pos uint32, s Slice) bool {
-			return pos == first.pos && s.Id == first.id && s.Off == first.off && s.Len == first.len
-		}
-		if !isFirst(pos, c[0]) {
-			// it's not the first slice, compact it
+		_, size1, _ := compactChunk(chunk[skipped+1:])
+		reduced := size - size1
+		if size1 == 0 || reduced < chunk[skipped].len || reduced*5 < size || reduced < 2<<20 {
```

Comment:
We can also consider the saved space `chunk[skipped].size`

Reply:
Will it be over complicated since we have to check the slice reference to know whether the slice can be deleted?

Reply:
no. the slice will be deleted after all the refs are compacted